src/plugin/debugger/mhpdbg.c

```diff
@@ -257,104 +255,104 @@ static int mhp_input(void)
+    /* NOTE: if there is input on mhpdbg.fdin, as result of handle_signals
+     *       io_select() is called and this then calls mhp_input.
+     *       ( all clear ? )
+     */
+    if (mhpdbg.nbytes <= 0) {
+      if (traceloop && mhpdbgc.stopped) {
+        mhpdbg.nbytes = strlen(loopbuf);
+        memcpy(mhpdbg.recvbuf, loopbuf, mhpdbg.nbytes + 1);
+      } else {
+        if (mhpdbgc.stopped) {
+          usleep(JIFFIE_TIME / 10);
+          continue;
+        } else
+          break;
       }
-      if ((mhpdbg.recvbuf[0] == 'q') && (mhpdbg.recvbuf[1] <= ' ')) {
-	 if (mhpdbgc.stopped) {
-	   mhp_cmd("g");
-	   mhp_send();
-	 }
-	 mhpdbg.active = 0;
-	 mhpdbg.sendptr = 0;
-         mhpdbg.nbytes = 0;
-         break;
+    } else {
+      if (traceloop) {
+        traceloop = loopbuf[0] = 0;
       }
-      mhpdbg.recvbuf[mhpdbg.nbytes] = 0x00;
-      ptr = (char *)mhpdbg.recvbuf;
-      while (ptr && *ptr) {
-	ptr1 = strsep(&ptr, "\r\n");
-	if (!ptr1)
-	  ptr1 = ptr;
-	if (!ptr1)
-	  break;
-        mhp_cmd(ptr1);
+    }
+    if ((mhpdbg.recvbuf[0] == 'q') && (mhpdbg.recvbuf[1] <= ' ')) {
+      if (mhpdbgc.stopped) {
+        mhp_cmd("g");
         mhp_send();
       }
+      mhpdbg.active = 0;
+      mhpdbg.sendptr = 0;
       mhpdbg.nbytes = 0;
-   }
-   in_poll_loop--;
+      break;
+    }
+    mhpdbg.recvbuf[mhpdbg.nbytes] = 0x00;
+    ptr = (char *)mhpdbg.recvbuf;
+    while (ptr && *ptr) {
+      ptr1 = strsep(&ptr, "\r\n");
+      if (!ptr1)
+        ptr1 = ptr;
+      if (!ptr1)
+        break;
+      mhp_cmd(ptr1);
+      mhp_send();
+    }
+    mhpdbg.nbytes = 0;
+  }
+  in_poll_loop--;
 }
 
 static void mhp_pre_vm86(void)
 {
-    if (!mhpdbg.active)
-	return;
-    if ((!in_dpmi_pm() && isset_TF()) ||
-	    (in_dpmi_pm() && dpmi_mhp_issetTF())) {
-	if (mhpdbgc.trapip != mhp_getcsip_value()) {
-	    mhpdbgc.trapcmd = 0;
-	    mhpdbgc.stopped = 1;
-	    mhp_poll();
-	}
+  if (!mhpdbg.active)
+    return;
+  if ((!in_dpmi_pm() && isset_TF()) || (in_dpmi_pm() && dpmi_mhp_issetTF())) {
+    if (mhpdbgc.trapip != mhp_getcsip_value()) {
+      mhpdbgc.trapcmd = 0;
+      mhpdbgc.stopped = 1;
+      mhp_poll();
     }
-    if (mhpdbgc.stopped)
-	mhp_poll();
+  }
+  if (mhpdbgc.stopped)
+    mhp_poll();
 }
 
 static void mhp_poll(void)
 {
 
   if (!mhpdbg.active) {
-     mhpdbg.nbytes = 0;
-     return;
+    mhpdbg.nbytes = 0;
+    return;
   }
 
   if (mhpdbg.active == 1) {
     /* new session has started */
     mhpdbg.active++;
 
-    mhp_printf ("%s", mhp_banner);
+    mhp_printf("%s", mhp_banner);
     mhp_poll_loop();
   }
   if (mhpdbgc.want_to_stop) {
     mhpdbgc.stopped = 1;
     mhpdbgc.want_to_stop = 0;
   }
   if (mhpdbgc.stopped) {
-      if (dosdebug_flags & DBGF_LOG_TEMPORARY) {
-         dosdebug_flags &= ~DBGF_LOG_TEMPORARY;
-	 mhp_cmd("log off");
-      }
-      mhp_cmd("r0");
-      mhp_send();
+    if (dosdebug_flags & DBGF_LOG_TEMPORARY) {
+      dosdebug_flags &= ~DBGF_LOG_TEMPORARY;
+      mhp_cmd("log off");
```

Comment:
Its still inconsistent.

Reply:
funny thing is that I can't see anything wrong with either of these, can you specific?

Reply:
Then you need to use editor
that shows tabs. You have some
lines tabbed, some spaced.

Reply:
I tweaked my vim config to show `⇒⇒` for tabs, and disabled tab for 'c' and 'h' files now so I can be sure I didn't insert any.